core/src/main/kotlin/utilities/json.kt

```diff
@@ -0,0 +1,61 @@
+package org.jetbrains.dokka.utilities
+
+import com.fasterxml.jackson.core.JsonGenerator
+import com.fasterxml.jackson.databind.*
+import com.fasterxml.jackson.databind.jsonFormatVisitors.JsonFormatVisitorWrapper
+import com.fasterxml.jackson.databind.module.SimpleModule
+import com.fasterxml.jackson.databind.ser.std.StdScalarSerializer
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.jacksonTypeRef
+import java.io.File
+import java.lang.reflect.Type
+import com.fasterxml.jackson.core.type.TypeReference as JacksonTypeReference
+
+private val objectMapper = run {
+    val module = SimpleModule().apply {
+        addSerializer(FileSerializer)
+    }
+    jacksonObjectMapper()
+        .registerModule(module)
+        .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+        .configure(DeserializationFeature.ACCEPT_SINGLE_VALUE_AS_ARRAY, true)
+}
+
+@PublishedApi
+internal class TypeReference<T> private constructor(
+    internal val jackson: JacksonTypeReference<T>
+) {
+    companion object {
+        @OptIn(ExperimentalStdlibApi::class)
+        internal inline operator fun <reified T> invoke(): TypeReference<T> = TypeReference(jacksonTypeRef())
+    }
+}
+
+@PublishedApi
+internal fun toJsonString(value: Any): String = objectMapper.writeValueAsString(value)
+
+@PublishedApi
+internal inline fun <reified T : Any> parseJson(json: String): T {
+    return parseJson(json, TypeReference())
+}
+
+@PublishedApi
+internal fun <T : Any> parseJson(json: String, typeReference: TypeReference<T>): T {
+    return objectMapper.readValue(json, typeReference.jackson)
+}
+
+
+object FileSerializer : StdScalarSerializer<File>(File::class.java) {
+    override fun serialize(value: File, g: JsonGenerator, provider: SerializerProvider) {
+        g.writeString(value.path)
```

Comment:
I think this follows the proposal of @SUPERCILEX , but will break the existing format which uses something like 
`"path": "path/to/file"` as json format 🤔